packages/graphql-codegen-cli/tests/codegen.spec.ts

```diff
@@ -928,6 +928,27 @@ describe('Codegen Executor', () => {
         expect(e.message).toContain('Failed to load custom loader');
       }
     });
+
+    it('Should allow loading docs by require+config method', async () => {
+      await executeCodegen({
+        schema: ['./tests/test-documents/schema.graphql'],
+        documents: [
+          {
+            require: './tests/test-documents/valid.graphql',
```

Comment:
The goal was actually to do it the other way around. 

So you can use a loader, and pass config to it: 

```
require: ./tests/custom-loaders/custom-documents-loader.js
config:
   loadFilesFrom: "./tests/test-documents/valid.graphql"
```